src/juliaDebug.ts

```diff
@@ -266,48 +261,35 @@ export class JuliaDebugSession extends LoggingDebugSession {
 	        connectedPromise.notify()
 	    })
 
-	    const serverForWrapper = net.createServer(socket => {
-	        this._debuggeeWrapperSocket = socket
-	    })
-
-	    serverForWrapper.listen(pnForWrapper, () => {
-	        serverForWrapperPromise.notify()
-	    })
-
-	    await serverForWrapperPromise.wait()
-
 	    server.listen(pn, () => {
 	        serverListeningPromise.notify()
 	    })
 
 	    await serverListeningPromise.wait()
 
-	    this._debuggeeTerminal = vscode.window.createTerminal({
-	        name: 'Julia Debugger',
-	        shellPath: this._juliaPath,
-	        shellArgs: [
-	            '--color=yes',
-	            '--startup-file=no',
-	            '--history-file=no',
-	            join(this._context.extensionPath, 'scripts', 'debugger', 'launch_wrapper.jl'),
-	            pn,
-	            pnForWrapper,
-	            args.cwd,
-	            args.juliaEnv,
-	            getCrashReportingPipename()
-	        ],
-	        env: {
-	            JL_ARGS: args.args ? args.args.map(i => Buffer.from(i).toString('base64')).join(';') : ''
-	        }
-	    })
-	    this._debuggeeTerminal.show(false)
-	    const asdf: Array<Disposable> = []
-	    vscode.window.onDidCloseTerminal((terminal) => {
-	        if (terminal === this._debuggeeTerminal) {
-	            this.sendEvent(new TerminatedEvent())
-	            asdf[0].dispose()
-	        }
-	    }, this, asdf)
+	    const foo = args.args ? args.args.map(i => Buffer.from(i).toString('base64')).join(';') : ''
+
+	    this.runInTerminalRequest(
+	        {
+	            kind: 'integrated',
+	            // kind: 'external',
+	            title: 'Julia Debugger',
+	            cwd: args.cwd,
+	            args: [
+	                this._juliaPath,
+	                '--color=yes',
+	                '--history-file=no',
+	                '--startup-file=no',
+	                `--project=${args.juliaEnv}`,
+	                join(this._context.extensionPath, 'scripts', 'debugger', 'run_debugger.jl'),
```

Comment:
as I reported in other comments, having the `launch_wrapper.jl` file allows us to add custom options when we start de debugger, for instance debugging with a sysimage (julia --sysimage path_to_sysimage). With the current solution we are limited to the options presented here (color, history-file etc). We shall have this `launch_wrapper.jl`  back or we shall have a way to specify the extra arguments to julia.

Reply:
https://github.com/julia-vscode/julia-vscode/issues/2552